spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ReRegistrationPredicate.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.consul.discovery;
+
+import com.ecwid.consul.v1.OperationException;
+
+/**
+ * Predicate on whether to re-register service.
+ *
+ * @author Toshiaki Maki
+ */
+public interface ReRegistrationPredicate {
```

Comment:
Why new interface? Just use `Predicate<OperationException`.

Reply:
I tend to avoid using classes under `java.*` package as a bean type. It can potentially conflict with other libraries. For example, Spring Cloud Function uses `Function`, `Consumer` and `Supplier`. If there is no possibility of a conflict in `Predicate`, I don't mind dropping this class.